src/main/java/org/javarosa/xpath/XPathConditional.java

```diff
@@ -51,6 +51,10 @@ public Object evalRaw(DataInstance model, EvaluationContext evalContext) {
             if (xpath != null) {
                 throw new XPathUnsupportedException(xpath);
             } else {
+                String contextMessage = String.format(" Question with id: \'%s\' has issue with condition expression: \'%s\'",
+                        evalContext.getContextRef().getNameLast(), expr);

```

Comment:
Couple of small notes:

1) It's preferable to use the full context ref for these, rather than drilling down to the last qname. /data/something/myquestion is the easiest way to track that. 

2) ContextRef can be a somewhat sketchy indicator of where an conditional is calculating, since we simplify the graph of triggerables, so I'd probably structure the message such that the XPathExpression itself is the first thing the user sees, since it'll be more canonically useful, and I might be a bit less concrete in the message, ie: ("Error calculating expression: "expression", being calculated for "context")

3) With user facing XPathExpressions, it's good to use toPrettyString() to format the output if the raw text isn't available.

4) "Conditional" is a useful concept for developer facing messages, but will be confusing for end users. If it's known whether an expression is a "calculate" or a "Display Condition", it's fine to use those two words, but otherwise best to leave out what type of expression was having an error

5) For tickets like this it's very helpful on review to add what the Replication Condition now shows as a result of the change, can you add that to the PR summary page?

Reply:
@ctsims  toPrettyString() can't be used because there could be an exception during parse boolean expression, this was the case with test expression in issue reproducing app.